From 0a8b16283b51ce0d8e34197052f35b9e0de7f7b5 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Wed, 26 Aug 2015 16:54:23 +0800
Subject: [PATCH 0297/5242] MLK-11424: dmaengine: imx-sdma: slave_sg can't
 work after running cyclic

commit  47cb2707cd5a2d9c343fed186a99da12e86047fe from
https://source.codeaurora.org/external/imx/linux-imx.git

After running asrc p2p, then asrc m2m can't work. It is caused by the
sdmac->flags isn't reset after running. then sdma_int_handler go to
wrong branch.

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/dma/imx-sdma.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/dma/imx-sdma.c b/drivers/dma/imx-sdma.c
index d3fa138..f0bacce 100644
--- a/drivers/dma/imx-sdma.c
+++ b/drivers/dma/imx-sdma.c
@@ -1340,6 +1340,7 @@ static int sdma_transfer_init(struct sdma_channel *sdmac,
 	sdmac->buf_tail = 0;
 	sdmac->flags = 0;
 	sdmac->direction = direction;
+	sdmac->flags = 0;
 
 	ret = sdma_load_context(sdmac);
 	if (ret)
-- 
1.7.9.5

